### PR TITLE
New task list loading placeholder for experiment 1

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Made two different changes:
- Some general changes to improve the loading time of the task list
- Make use of the Two column task list placeholder for the first task list experiment 

task-list-placeholder-experiment1

### How to test the changes in this Pull Request:

1. On a new site finish the onboarding wizard
2. Load the home screen
3. It should show a new task list placeholder that actually shows a header as part of the loading design (see GIF as well)
4. Task list should display and load before the `notes` have been loaded below

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/tasks/fills/purchase.js [hunk 1]
```diff
@@ -47,31 +47,38 @@ const PurchaseTaskItem = () => {
...
-						<CartModal
-							onClose={ () => toggleCartModal() }
-							onClickPurchaseLater={ () => toggleCartModal() }
-						/>
-					) }
-				</>
+			{ ( { defaultTaskItem } ) => (
+				<PurchaseTaskItem defaultTaskItem={ defaultTaskItem } />
 			) }
 		</WooOnboardingTaskListItem>
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/tasks/fills/purchase.js:77]: This is a SlotFill which was retrieving all of it's data without being displayed yet. So I moved that logic out in it's own component, that way we don't make any unnecessary requests when the homescreen first loads.

File: plugins/woocommerce-admin/client/tasks/tasks.tsx [hunk 2]
```diff
@@ -15,29 +15,20 @@ import { recordEvent } from '@woocommerce/tracks';
...
-const TaskList = lazy(
-	() => import( /* webpackChunkName: "task-list" */ './task-list' )
-);
-
-const TwoColumnTaskList = lazy(
-	() =>
-		import(
-			/* webpackChunkName: "two-column-task-list" */ '../two-column-tasks/task-list'
-		)
-);
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/tasks/tasks.tsx:36]: Using chunks for the task list was actually causing a delay, as retrieving the chunks got stuck behind some other requests that took longer.

File: plugins/woocommerce-admin/client/two-column-tasks/index.js [hunk 3]
```diff
@@ -14,25 +14,7 @@ import TaskList from './task-list';
 import TaskListPlaceholder from './placeholder';
 import { Task } from '../tasks/task';
 
-const taskDashboardSelect = ( select ) => {
-	const { getOption, hasFinishedResolution } = select( OPTIONS_STORE_NAME );
-
-	return {
-		keepCompletedTaskList: getOption(
-			'woocommerce_task_list_keep_completed'
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/two-column-tasks/index.js:22]: Moved this option to the back end, as an added TaskList param.

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php [hunk 4]
```diff
@@ -373,4 +374,17 @@ function ( $all, $curr ) {
...
+
+	/**
+	 * Add visible list ids to component settings.
+	 *
+	 * @param array $settings Component settings.
+	 *
+	 * @return array
+	 */
+	public static function task_list_preloaded_settings( $settings ) {
+		$settings['visibleTaskListIds'] = array_keys( self::get_visible() );
```
Comment (louwie17) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php:386]: Set the task list ids so we can actually show the correct loading component.

File: plugins/woocommerce-admin/client/tasks/tasks.tsx [hunk 5]
```diff
@@ -56,9 +57,9 @@ export const Tasks: React.FC< TasksProps > = ( { query } ) => {
 
 	const { isResolving, taskLists } = useSelect( ( select ) => {
 		return {
-			isResolving: select( ONBOARDING_STORE_NAME ).isResolving(
-				'getTaskLists'
-			),
+			isResolving: ! select(
+				ONBOARDING_STORE_NAME
+			).hasFinishedResolution( 'getTaskLists' ),
```
Comment (joelclimbsthings) [inline plugins/woocommerce-admin/client/tasks/tasks.tsx:62]: 👍🏻

**Review by joelclimbsthings — APPROVED** [review]
Thanks for this @louwie17 , everything worked as expected and the code looks good. 

Not a requirement, but we do have some typescript issues in `tasks.tsx` that are likely not related to this PR, but perhaps should be addressed at some point. We also have this line:
`const taskListsFinished = false;`
which is just an orphaned `const`, so I'm not sure if that's a placeholder for something further?

All that said, consider that all optional and I'm preapproving 🚢

**Comment by louwie17** [comment]
> Not a requirement, but we do have some typescript issues in `tasks.tsx` that are likely not related to this PR, but perhaps should be addressed at some point. We also have this line: `const taskListsFinished = false;` which is just an orphaned `const`, so I'm not sure if that's a placeholder for something further?

Good finds on these, I might have those addressed in a separate PR

**Pull request merged**